devdoc/constbuffer_array_requirements.md

```diff
@@ -98,6 +98,34 @@ Note: `constbuffer_array_create_from_buffer_index_and_count` does not increment
 
 **SRS_CONSTBUFFER_ARRAY_42_016: [** If any error occurs then `constbuffer_array_create_from_buffer_index_and_count` shall fail and return `NULL`. **]**
 
+### constbuffer_array_create_from_start_and_end
+
+```c
+MOCKABLE_FUNCTION(, CONSTBUFFER_ARRAY_HANDLE, constbuffer_array_create_from_start_and_end, CONSTBUFFER_ARRAY_HANDLE, original, uint32_t, start_buffer_index, uint32_t, start_buffer_offset, uint32_t, start_buffer_size, uint32_t, end_buffer_index, uint32_t, end_buffer_offset, uint32_t, end_buffer_size);
+```
+
+`constbuffer_array_create_from_start_and_end` creates a new const buffer array which is a subset of the existing array in `original`.
+
+**SRS_CONSTBUFFER_ARRAY_07_001: [** If `original` is `NULL` then `constbuffer_array_create_from_start_and_end` shall fail and return `NULL`.  **]**
+
+**SRS_CONSTBUFFER_ARRAY_07_002: [** If `start_buffer_index` is greater than the number of buffers in `original` then `constbuffer_array_create_from_start_and_end` shall fail and return `NULL`.  **]**
+
+**SRS_CONSTBUFFER_ARRAY_07_003: [** If `end_buffer_index` is greater than the number of buffers in `original` then `constbuffer_array_create_from_start_and_end` shall fail and return `NULL`.  **]**
+
+**SRS_CONSTBUFFER_ARRAY_07_004: [** If `start_buffer_index` is greater than or equal to `end_buffer_index` then `constbuffer_array_create_from_start_and_end` shall fail and return `NULL`. **]**
+
+**SRS_CONSTBUFFER_ARRAY_07_005: [** `constbuffer_array_create_from_start_and_end` shall get the subset of the `start_buffer` and `end_buffer` by calling `CONSTBUFFER_CreateFromOffsetAndSize`. **]**
```

Comment:
>`start_buffer` and `end_buffer` [](http://example.com/codeflow?start=110&length=31)

I think this should be 2 specs (well, probably 3)

- shall get the first buffer by calling `CONSTBUFFER_CreateFromOffsetAndSize` with `start_buffer_offset`
- shall get the last buffer by calling `CONSTBUFFER_CreateFromOffsetAndSize` with `end_buffer_size`
- If `buffer_count` is 1 then shall get the only buffer by calling `CONSTBUFFER_CreateFromOffsetAndSize` with `start_buffer_offset` and `end_buffer_size`

Reply:
I guess if you remove the start_buffer_size parameter, then there is also a spec that says this will compute the first buffer size (as the total buffer size minus the start offset).

Reply:
added